src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -20,151 +20,123 @@
         _ = rules ?? throw new ArgumentNullException(nameof(rules));
 
-        var rulesetPath = CreateRuleSet(language, rules, false);
-        var deactivatedRulesetPath = CreateRuleSet(language, rules, true);
-        var analyzerPlugins = FetchAnalyzerPlugins(language, rules.Where(x => x.IsActive));
-        var additionalFiles = WriteAdditionalFiles(language, rules.Where(x => x.IsActive));
+        var rulesetPath = CreateRuleSet(rules, false);
+        var deactivatedRulesetPath = CreateRuleSet(rules, true);
+        var analyzerPlugins = FetchAnalyzerPlugins(rules.Where(x => x.IsActive));
+        var additionalFiles = WriteAdditionalFiles(rules.Where(x => x.IsActive));
 
         return new AnalyzerSettings(language, rulesetPath, deactivatedRulesetPath, analyzerPlugins, additionalFiles);
     }
 
-    private string CreateRuleSet(string language, IEnumerable<SonarRule> rules, bool deactivateAll)
+    private string CreateRuleSet(IEnumerable<SonarRule> rules, bool deactivateAll)
     {
-        var ruleSetGenerator = new RoslynRuleSetGenerator(this.sonarProperties, deactivateAll);
+        var ruleSetGenerator = new RoslynRuleSetGenerator(sonarProperties, deactivateAll);
         var ruleSet = ruleSetGenerator.Generate(language, rules);
-        Debug.Assert(ruleSet != null, "Expecting the RuleSet to be created.");
-        Debug.Assert(ruleSet.Rules != null, "Expecting the RuleSet.Rules to be initialized.");
-
-        var rulesetFilePath = Path.Combine(this.teamBuildSettings.SonarConfigDirectory, string.Format(deactivateAll ? RulesetFileNameNone : RulesetFileNameNormal, language));
-        this.logger.LogDebug(Resources.RAP_UnpackingRuleset, rulesetFilePath);
-
+        var rulesetFilePath = Path.Combine(teamBuildSettings.SonarConfigDirectory, string.Format(deactivateAll ? RulesetFileNameNone : RulesetFileNameNormal, language));
+        logger.LogDebug(Resources.RAP_UnpackingRuleset, rulesetFilePath);
         ruleSet.Save(rulesetFilePath);
-
         return rulesetFilePath;
     }
 
-    private IEnumerable<string> WriteAdditionalFiles(string language, IEnumerable<SonarRule> activeRules)
-    {
-        Debug.Assert(activeRules != null, "Supplied active rules should not be null");
+    private IEnumerable<string> WriteAdditionalFiles(IEnumerable<SonarRule> activeRules) =>
+        TryWriteSonarLintXmlFile(activeRules, out var filePath) ? [filePath] : [];
 
-        if (TryWriteSonarLintXmlFile(language, activeRules, out var filePath))
-        {
-            Debug.Assert(File.Exists(filePath), "Expecting the additional file to exist: {0}", filePath);
-            return new[] { filePath };
-        }
-
-        return Enumerable.Empty<string>();
-    }
-
-    private bool TryWriteSonarLintXmlFile(string language, IEnumerable<SonarRule> activeRules, out string sonarLintXmlPath)
+    private bool TryWriteSonarLintXmlFile(IEnumerable<SonarRule> activeRules, out string sonarLintXmlPath)
```

Comment:
`out` parameters are generally a code smell. And since we have pattern matching, it's typically easy to get rid of them
```suggestion
    private string WriteSonarLintXmlFile(IEnumerable<SonarRule> activeRules)
```